docs/source/fitting/fitfunctions/MultiDomainFunction.rst

```diff
@@ -0,0 +1,86 @@
+.. _func-MultiDomainFunction:
+
+=======================
+Multi Domain Function
+=======================
+
+.. index:: MultiDomainFunction
+
+Description
+-----------
+
+A multidomain function includes information about
+which domain a particular entry (function) will be applied to.
+The members of the multidomain function are indexed from 0 to the number of functions minus 1.
+Multidomain functions do not have their own parameters, instead they use
+parameters of the member functions. Parameter names are formed from the
+member function's index and its parameter name: f[domain_index].[name]. For
+example, "f0.A0" would refer to the "A0" parameter of the first function in the multidomain function.
+
+The input string to create a multidomain function is formed by including
+the $domains=n directive in the function string. For example, the string for two
+:ref:`Gaussians <func-Gaussian>` is given by,
+
+``composite=MultiDomainFunction,NumDeriv=1;name=Gaussian,$domains=i,PeakCentre=0,Height=10,Sigma=0.7;name=Gaussian,$domains=i,PeakCentre=0.1,Height=10,Sigma=0.7``
+
+where we note that the multidomain function is itself a :ref:`Composite function <func-CompositeFunction>`. The inclusion of $domains=i specifies that the ith
+function will correspond to the ith domain of the data. As a concrete example, consider fitting two spectra (domains) of
+a workspace,
+
+.. code-block:: python
+
+    def linear(x):
+        return 1 + x
+
+    def quadratic(x):
+        return 1 + 3*x + 4*np.square(x)
+
+    x = np.linspace(-2, 2, 151)
+    y1 = linear(x)
+    y2 = quadratic(x)
+    ys = np.concatenate((y1, y2))
+
+    ws = CreateWorkspace(DataX=x, DataY=ys, NSpec=2)
+
+    multiFunc="composite=MultiDomainFunction,NumDeriv=1;name=Quadratic,$domains=1;name=LinearBackground,$domains=0"
+    func = FunctionFactory.Instance().createInitialized(multiFunc)
+    fit_output = Fit(Function=func,
+                     InputWorkspace=ws, WorkspaceIndex=0,
+                     InputWorkspace_1=ws, WorkspaceIndex_1=1,
+                     Output='fit')
+
+As $domains=1 is set for the Quadratic function, it will be used to fit the second domain of the input workspace,
+corresponding to WorkspaceIndex=1. Similarly, the LinearBackground will be used to fit the first domain of the workspace,
+WorkspaceIndex=0. If instead we had specified domains=i, the function would have been fit based on the order of creation,
+i.e. the quadratic function would fit to the first domain and the LinearBackground to the second domain.
```

Comment:
Nice explanation of `$domains=i`, would it be possible to explain what `$domains=All` means? I am guessing it means the composite is used for all domains?

Reply:
As you say, I think it means its used for all domains, but couldn't find anything describing its behaviour. I tested it out on a simple fit and that’s what it seems to do, so I've added that to the docs.